src/test/java/CvStageTest.java

```diff
@@ -0,0 +1,718 @@
+            assertEquals(new org.openpnp.model.Point(camera.getWidth()/2 + overridingParameter.getX(),
+                    camera.getHeight()/2 - overridingParameter.getY()), overriddenParameter);
+        }
+        
+        //To Area conversion
+        {
+            //From Area
+            CvPipeline pipeline = new CvPipeline();
+            pipeline.setProperty("camera", camera);
+            Area originalParameter = new Area(39.5, AreaUnit.SquareMillimeters);
+            Area overridingParameter = new Area(87.4, AreaUnit.SquareMillimeters);
+            Area overriddenParameter;
+            overriddenParameter = CvStage.getPossiblePipelinePropertyOverride(originalParameter, 
+                    pipeline, propName);
+            assertEquals(originalParameter, overriddenParameter);
+            pipeline.setProperty(propName, overridingParameter);
+            overriddenParameter = CvStage.getPossiblePipelinePropertyOverride(originalParameter, 
+                    pipeline, propName);
+            assertEquals(overridingParameter, overriddenParameter);
+        }
+
+        //To Length conversion
+        {
+            //From Length
+            CvPipeline pipeline = new CvPipeline();
+            pipeline.setProperty("camera", camera);
+            Length originalParameter = new Length(39.5, LengthUnit.Millimeters);
+            Length overridingParameter = new Length(87.4, LengthUnit.Millimeters);
+            Length overriddenParameter;
+            overriddenParameter = CvStage.getPossiblePipelinePropertyOverride(originalParameter, 
+                    pipeline, propName);
+            assertEquals(originalParameter, overriddenParameter);
+            pipeline.setProperty(propName, overridingParameter);
+            overriddenParameter = CvStage.getPossiblePipelinePropertyOverride(originalParameter, 
+                    pipeline, propName);
+            assertEquals(overridingParameter, overriddenParameter);
+        }
+
+        //To Location conversion
+        {
+            //From Location
+            CvPipeline pipeline = new CvPipeline();
+            pipeline.setProperty("camera", camera);
+            Location originalParameter = new Location(LengthUnit.Millimeters, 39.5, 57.2, -34.5, 93.3);
+            Location overridingParameter = new Location(LengthUnit.Millimeters, 87.4, -38.4, 84.2, 18.4);
+            Location overriddenParameter;
+            overriddenParameter = CvStage.getPossiblePipelinePropertyOverride(originalParameter, 
+                    pipeline, propName);
+            assertEquals(originalParameter, overriddenParameter);
+            pipeline.setProperty(propName, overridingParameter);
+            overriddenParameter = CvStage.getPossiblePipelinePropertyOverride(originalParameter, 
+                    pipeline, propName);
+            assertEquals(overridingParameter, overriddenParameter);
+        }
+
+    }
+
+    static class TestCamera extends AbstractHeadMountable implements Camera {
```

Comment:
Hi @tonyluken,

could this CvStageTest.TestCamera be replaced by or inherit from VisionUtilsTest.TestCamera? It looks identical.

I'm just asking, because these mock-ups are a nuisance to maintain. 

_Mark

Reply:
Hi @markmaker,

Yes! and done.